WcaOnRails/app/models/user.rb

```diff
@@ -1375,4 +1379,89 @@ def can_access_wfc_senior_matters?
   def can_edit_translators?
     can_edit_any_groups? || software_team?
   end
+
+  private def board_role
+    {
+      group: {
+        id: 'board',
+        name: 'WCA Board of Directors',
+        group_type: UserGroup.group_types[:board],
+        is_hidden: false,
+        is_active: true,
+        metadata: {
+          friendly_id: 'board',
+        },
+      },
+      is_active: true,
+      user: self,
+      metadata: {
+        status: 'member',
+      },
+    }
+  end
+
+  def roles(logged_in_user = nil)
```

Comment:
Now the method is clear, but I am not happy that it does two things at once.
It computes the `roles` _and_ it also filters them based on some login condition.

I feel quite strongly that the user model itself shouldn't know anything about "users accessing other users' information" and similar conditions. Have this method return a definitive list, without any filtering, and then do whatever filtering may be required at the place wherever `user.roles` is being called

Reply:
Also worth noting that there is `private def filter_roles_for_logged_in_user` in `user_roles_controller`, which sounds like the perfect compromise